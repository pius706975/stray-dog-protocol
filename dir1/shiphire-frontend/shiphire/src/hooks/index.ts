export * from './useAuth';
export * from './useRenter';
export * from './useShip';
export * from './useUser';
export * from './useHelper';
export * from './useShipOwner';
export * from './useAdmin';
export * from './useMutationWithErrorHandler';
