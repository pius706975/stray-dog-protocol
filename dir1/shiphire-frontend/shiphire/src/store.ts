import { configureStore } from '@reduxjs/toolkit';
import {
    addShipSlice,
    documentNameSlice,
    modalSlice,
    networkErrorSlice,
    notifBadgeSlice,
    progressIndicatorSlice,
    renterPreferenceSlice,
    requestForQuoteSlice,
    shipSpecSlice,
    termsStatusSlice,
    userLanguageSlice,
    userLocationSlice,
    userStatusSlice,
} from './slices';

const store = configureStore({
    reducer: {
        modal: modalSlice.reducer,
        progressIndicator: progressIndicatorSlice.reducer,
        renterPreference: renterPreferenceSlice.reducer,
        requestForQuote: requestForQuoteSlice.reducer,
        userStatus: userStatusSlice.reducer,
        addShip: addShipSlice.reducer,
        documentName: documentNameSlice.reducer,
        termsStatus: termsStatusSlice.reducer,
        userLanguage: userLanguageSlice.reducer,
        userLocation: userLocationSlice.reducer,
        shipSpecSlice: shipSpecSlice.reducer,
        notifBadgeSlice: notifBadgeSlice.reducer,
        networkError: networkErrorSlice.reducer,
    },
});

export default store;
