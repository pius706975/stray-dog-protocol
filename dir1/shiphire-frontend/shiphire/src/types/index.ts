export * from './Components'
export * from './Navigators'
export * from './ReduxToolkit'
export * from './Screens'
export * from './Services'