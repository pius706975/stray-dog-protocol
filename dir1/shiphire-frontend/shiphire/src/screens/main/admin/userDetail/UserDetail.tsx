import React, { useCallback, useEffect, useMemo, useState } from 'react';
import { DetailUserProps } from '../../../../types';
import { Avatar, View } from 'react-native-ui-lib';
import {
    CustomText,
    NavigationKebabMenu,
    ScreenLayout,
} from '../../../../components';
import { Alert, Button, Platform } from 'react-native';
import CustomButton from '../../../../components/Button';
import {
    CircleCheckIcon,
    CircleInactiveIcon,
    Color,
} from '../../../../configs';
import {
    useUpdateUserActivation,
    useVerifyUserPhoneNumber,
} from '../../../../hooks';
import { useDispatch } from 'react-redux';
import { modalSlice } from '../../../../slices';
import { handleAxiosError } from '../../../../utils';
import { useTranslation } from 'react-i18next';
import { ModalConfirmation } from './component';

const UserDetail: React.FC<DetailUserProps> = ({ navigation, route }) => {
    const {
        _id,
        name,
        email,
        phoneNumber,
        roles,
        isActive,
        shipOwnerId,
        renterId,
        isPhoneVerified,
        isVerified,
    } = route.params.user;
    const [active, setActive] = React.useState<boolean>(isActive);
    const [isUserPhoneVerified, setIsUserPhoneVerified] =
        React.useState<boolean>(isPhoneVerified);
    const { t } = useTranslation('usermanagement');
    const plusPaddingV = Platform.OS === 'ios' ? 40 : 20;
    const [isSubmitting, setIsSubmitting] = useState(false);
    const mutationUpdateUserActivation = useUpdateUserActivation();
    const mutationVerifyUserPhoneNumber = useVerifyUserPhoneNumber();
    const { hideModal, showModal } = modalSlice.actions;
    const dispatch = useDispatch();
    const [visible, setvisible] = React.useState<boolean>(false);
    const [isVerifyPhoneModal, setIsVerifyPhoneModal] =
        useState<boolean>(false);
    const [isVerifyPhoneLoading, setIsVerifyPhoneLoading] =
        useState<boolean>(false);
    let roleText: string = '';
    if (roles === 'renter') {
        roleText = t('renter');
    } else if (roles === 'shipOwner') roleText = t('shipOwner');

    const handleShowModal = useCallback(() => {
        setvisible(prevVisible => !prevVisible);
    }, []);

    const handleVerifyPhoneModal = useCallback(() => {
        setIsVerifyPhoneModal(
            prevIsVerifyPhoneModal => !prevIsVerifyPhoneModal,
        );
    }, []);

    const updateActivation = useCallback(() => {
        setIsSubmitting(true);
        mutationUpdateUserActivation.mutate(
            { _id, isActive: !active },
            {
                onSuccess: resp => {
                    setIsSubmitting(false);
                    handleShowModal();
                    setActive(resp.data.data.isActive);

                    dispatch(
                        showModal({
                            status: 'success',
                            text: t('successMessageUpdateActivation'),
                        }),
                    );
                    setTimeout(() => {
                        dispatch(hideModal());
                    }, 4000);
                },
                onError: err => {
                    handleAxiosError(err);
                },
            },
        );
    }, [
        _id,
        active,
        mutationUpdateUserActivation,
        dispatch,
        handleShowModal,
        t,
    ]);

    const verifyUserPhoneNumber = useCallback(() => {
        setIsVerifyPhoneLoading(true);

        mutationVerifyUserPhoneNumber.mutate(
            { _id, isPhoneVerified: !isUserPhoneVerified },
            {
                onSuccess: resp => {
                    handleVerifyPhoneModal();
                    setIsUserPhoneVerified(resp.data.data.isPhoneVerified);

                    dispatch(
                        showModal({
                            status: 'success',
                            text: t('successVerifyUserPhoneNumber'),
                        }),
                    );
                    setTimeout(() => {
                        dispatch(hideModal());
                    }, 4000);
                },
                onError: err => {
                    handleAxiosError(err);
                },
                onSettled: () => {
                    setIsVerifyPhoneLoading(false);
                },
            },
        );
    }, [
        _id,
        isUserPhoneVerified,
        mutationVerifyUserPhoneNumber,
        dispatch,
        handleVerifyPhoneModal,
        t,
    ]);

    const menuOptions = useMemo(() => {
        const options = [
            {
                label: active ? t('inactiveUser') : t('activateUser'),
                action: handleShowModal,
            },
        ];

        if (!isUserPhoneVerified) {
            options.push({
                label: t('verifyPhoneNumber'),
                action: handleVerifyPhoneModal,
            });
        }

        return options;
    }, [
        active,
        handleShowModal,
        handleVerifyPhoneModal,
        isUserPhoneVerified,
        t,
    ]);

    useEffect(() => {
        navigation.setOptions({
            headerRight: () => <NavigationKebabMenu options={menuOptions} />,
        });
    }, [navigation, menuOptions]);

    return (
        <ScreenLayout backgroundColor="light" testId="adminUserDetail">
            <View
                paddingV-12
                paddingH-12
                paddingB-8
                style={{
                    backgroundColor: Color.primaryColor,
                    borderBottomColor: Color.primaryColor,
                    borderBottomWidth: 1,
                }}
                center>
                <Avatar name={name} useAutoColors size={100} />
                <CustomText
                    fontSize="xl2"
                    fontFamily="bold"
                    color="lightTextColor">
                    {name}
                </CustomText>
                <CustomText
                    fontSize="md"
                    fontFamily="regular"
                    color="lightTextColor">
                    {roleText}
                </CustomText>
            </View>
            <View paddingH-8 paddingT-8 flex-1>
                <View marginB-8>
                    <View row style={{ alignItems: 'center' }}>
                        <CustomText
                            fontSize="xl"
                            fontFamily="bold"
                            color="primaryColor">
                            Email
                        </CustomText>
                        <View marginL-5>
                            {isVerified ? (
                                <CircleCheckIcon />
                            ) : (
                                <CircleInactiveIcon />
                            )}
                        </View>
                    </View>
                    <CustomText
                        fontSize="md"
                        fontFamily="regular"
                        color="darkTextColor">
                        {email}
                    </CustomText>
                </View>
                <View marginB-8>
                    <View row style={{ alignItems: 'center' }}>
                        <CustomText
                            fontSize="xl"
                            fontFamily="bold"
                            color="primaryColor">
                            {t('phoneNumber')}
                        </CustomText>
                        <View marginL-5>
                            {isPhoneVerified ? (
                                <CircleCheckIcon />
                            ) : (
                                <CircleInactiveIcon />
                            )}
                        </View>
                    </View>
                    <CustomText
                        fontSize="md"
                        fontFamily="regular"
                        color="darkTextColor">
                        {phoneNumber ? phoneNumber : t('phoneNumberNotSubmit')}
                    </CustomText>
                </View>
                {renterId?.company || shipOwnerId?.company ? (
                    <>
                        <View marginB-8>
                            <CustomText
                                fontSize="xl"
                                fontFamily="bold"
                                color="primaryColor">
                                {t('companyName')}
                            </CustomText>
                            <CustomText
                                fontSize="md"
                                fontFamily="regular"
                                color="darkTextColor">
                                {shipOwnerId?.company.name ||
                                    renterId?.company.name}
                            </CustomText>
                        </View>
                        <View marginB-8>
                            <CustomText
                                fontSize="xl"
                                fontFamily="bold"
                                color="primaryColor">
                                {t('companyAddress')}
                            </CustomText>
                            <CustomText
                                fontSize="md"
                                fontFamily="regular"
                                color="darkTextColor">
                                {shipOwnerId?.company.address ||
                                    renterId?.company.address}
                            </CustomText>
                        </View>
                    </>
                ) : (
                    <View></View>
                )}
            </View>
            <ModalConfirmation
                visible={visible}
                onClose={handleShowModal}
                onSubmit={updateActivation}
                isSubmitting={isSubmitting}
                activeStatus={active}
            />
            <ModalConfirmation
                visible={isVerifyPhoneModal}
                onClose={handleVerifyPhoneModal}
                onSubmit={verifyUserPhoneNumber}
                isSubmitting={isVerifyPhoneLoading}
                activeStatus={active}
                customButtonText={t('verify')}
                customBodyText={t('verifyPhoneConfirmation')}
            />
        </ScreenLayout>
    );
};

export default UserDetail;
