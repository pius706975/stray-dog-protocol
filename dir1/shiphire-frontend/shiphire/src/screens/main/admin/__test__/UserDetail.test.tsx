import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { QueryClient, QueryClientProvider } from 'react-query';
import { MainAdminStackParamList } from '../../../../types';
import httpRequest from '../../../../services/api';
import MockAdapter from 'axios-mock-adapter';
import { Provider } from 'react-redux';
import store from '../../../../store';
import { NavigationContainer } from '@react-navigation/native';
import { UserDetail } from '../userDetail';
import { fireEvent, render, waitFor } from '@testing-library/react-native';
import { act } from 'react-test-renderer';
import { useUpdateUserActivation } from '../../../../hooks';

jest.mock('../../../../hooks');
jest.mock('react-i18next', () => ({
    // this mock makes sure any components using the translate hook can use it without a warning being shown
    useTranslation: () => {
        return {
            t: str => str,
            i18n: {
                changeLanguage: () => new Promise(() => {}),
            },
        };
    },
    initReactI18next: {
        type: '3rdParty',
        init: () => {},
    },
}));
const queryClient = new QueryClient();
const MainAdminStack = createNativeStackNavigator<MainAdminStackParamList>();

const mock = new MockAdapter(httpRequest);
const mockUserData = {
    _id: '123',
    name: 'Alma',
    email: 'alma@mail.com',
    phoneNumber: '0812345674',
    roles: 'renter',
    isActive: true,
    renterId: {
        _id: '123',
        company: {
            name: 'alma company',
            address: 'pahlawan street',
        },
    },
    shipOwnerId: {
        _id: '123',
        company: {
            name: 'alma company',
            address: 'pahlawan street',
        },
    },
    isVerified: true,
    isPhoneVerified: true,
    googleId: '12232323343',
};

const adminStackMockComponent = (
    <QueryClientProvider client={queryClient}>
        <Provider store={store}>
            <NavigationContainer>
                <MainAdminStack.Navigator>
                    <MainAdminStack.Screen
                        name="UserDetail"
                        component={UserDetail}
                        initialParams={{ user: mockUserData }}
                    />
                </MainAdminStack.Navigator>
            </NavigationContainer>
        </Provider>
    </QueryClientProvider>
);

describe('Testing user detail screen', () => {
    describe('Snapshot testing', () => {
        it('should render user detail screen correctly', () => {
            const userDetailScreen = render(adminStackMockComponent);
            expect(userDetailScreen).toMatchSnapshot();
        });
    });
    describe('Unit Testing', () => {
        it('should showing confimation modal', () => {
            const { getByTestId } = render(adminStackMockComponent);
            const showModal = getByTestId('activate-button');

            act(() => {
                fireEvent.press(showModal);
            });
            expect('updateConfirmation').toBeDefined();
        });
        it('should update user activation', async () => {
            mock.onPost('admin/activate-user').reply(200, {
                status: 'success',
            });
            (useUpdateUserActivation as jest.Mock).mockImplementation(() => ({
                mutate: jest.fn().mockImplementation((data, options) => {
                    const { onSuccess } = options;
                    const mockResponseData = {
                        data: {
                            data: {
                                isActive: false,
                            },
                        },
                    };
                    onSuccess(mockResponseData);
                }),
            }));

            const { getByTestId, getByText } = render(adminStackMockComponent);
            const showModal = getByTestId('activate-button');

            fireEvent.press(showModal);

            jest.advanceTimersByTime(2000);
            const updateStatus = getByTestId('update-status');
            fireEvent.press(updateStatus);

            await waitFor(() => {
                const modalState = store.getState().modal;

                expect(modalState).toEqual({
                    visible: true,
                    status: 'success',
                    text: 'successMessageUpdateActivation',
                });
            });
        });
    });
});
