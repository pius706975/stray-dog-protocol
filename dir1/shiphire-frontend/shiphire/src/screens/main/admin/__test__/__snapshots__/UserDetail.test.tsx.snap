// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing user detail screen Snapshot testing should render user detail screen correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    {
      "flex": 1,
    }
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                undefined,
                undefined,
                {
                  "flex": 1,
                },
                undefined,
                {},
                {},
                {},
                {
                  "backgroundColor": "#f0f6fb",
                  "justifyContent": "space-between",
                  "position": "relative",
                },
              ]
            }
            testID="adminUserDetail"
          >
            <RCTScrollView
              contentContainerStyle={
                {
                  "alignItems": undefined,
                  "flex": undefined,
                  "flexGrow": 1,
                  "gap": undefined,
                  "justifyContent": undefined,
                  "marginBottom": undefined,
                  "padding": undefined,
                  "paddingVertical": undefined,
                }
              }
            >
              <View>
                <View
                  center={true}
                  paddingB-8={true}
                  paddingH-12={true}
                  paddingV-12={true}
                  style={
                    [
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      {
                        "paddingBottom": 8,
                        "paddingHorizontal": 12,
                        "paddingVertical": 12,
                      },
                      {},
                      {
                        "alignItems": "center",
                        "justifyContent": "center",
                      },
                      {
                        "backgroundColor": "#216dab",
                        "borderBottomColor": "#216dab",
                        "borderBottomWidth": 1,
                      },
                    ]
                  }
                >
                  <View
                    accessibilityLabel="Avatar"
                    accessibilityRole="image"
                    accessible={false}
                    style={
                      [
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        [
                          {
                            "alignItems": "center",
                            "borderRadius": 999,
                            "height": 100,
                            "justifyContent": "center",
                            "width": 100,
                          },
                          undefined,
                        ],
                      ]
                    }
                  >
                    <View
                      style={
                        [
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          [
                            {
                              "alignItems": "center",
                              "borderRadius": 999,
                              "bottom": 0,
                              "justifyContent": "center",
                              "left": 0,
                              "position": "absolute",
                              "right": 0,
                              "top": 0,
                            },
                            {
                              "backgroundColor": "#0F59E6",
                            },
                            false,
                          ],
                        ]
                      }
                      testID="undefined.container"
                    >
                      <Text
                        numberOfLines={1}
                        style={
                          [
                            {
                              "backgroundColor": "transparent",
                              "color": "#20303C",
                              "writingDirection": "ltr",
                            },
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            [
                              {
                                "fontFamily": "System",
                                "fontSize": 28,
                                "fontWeight": "800",
                                "lineHeight": 32,
                              },
                              {
                                "backgroundColor": "transparent",
                                "color": "#20303C",
                                "lineHeight": undefined,
                              },
                              {
                                "color": "#20303C",
                              },
                            ],
                          ]
                        }
                        testID="undefined.label"
                      >
                        A
                      </Text>
                    </View>
                  </View>
                  <Text
                    numberOfLines={0}
                    style={
                      {
                        "borderBottomWidth": 0,
                        "borderColor": undefined,
                        "color": "#f0f6fb",
                        "fontFamily": "Kanit-Bold",
                        "fontSize": 26,
                        "lineHeight": undefined,
                        "textAlign": undefined,
                      }
                    }
                  >
                    Alma
                  </Text>
                  <Text
                    numberOfLines={0}
                    style={
                      {
                        "borderBottomWidth": 0,
                        "borderColor": undefined,
                        "color": "#f0f6fb",
                        "fontFamily": "Kanit-Regular",
                        "fontSize": 18,
                        "lineHeight": undefined,
                        "textAlign": undefined,
                      }
                    }
                  >
                    renter
                  </Text>
                </View>
                <View
                  flex-1={true}
                  paddingH-8={true}
                  paddingT-8={true}
                  style={
                    [
                      undefined,
                      undefined,
                      {
                        "flex": 1,
                      },
                      undefined,
                      {
                        "paddingHorizontal": 8,
                        "paddingTop": 8,
                      },
                      {},
                      {},
                      undefined,
                    ]
                  }
                >
                  <View
                    marginB-8={true}
                    style={
                      [
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        {},
                        {
                          "marginBottom": 8,
                        },
                        {},
                        undefined,
                      ]
                    }
                  >
                    <View
                      row={true}
                      style={
                        [
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          {},
                          {},
                          {
                            "flexDirection": "row",
                          },
                          {
                            "alignItems": "center",
                          },
                        ]
                      }
                    >
                      <Text
                        numberOfLines={0}
                        style={
                          {
                            "borderBottomWidth": 0,
                            "borderColor": undefined,
                            "color": "#216dab",
                            "fontFamily": "Kanit-Bold",
                            "fontSize": 22,
                            "lineHeight": undefined,
                            "textAlign": undefined,
                          }
                        }
                      >
                        Email
                      </Text>
                      <View
                        marginL-5={true}
                        style={
                          [
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            {},
                            {
                              "marginLeft": 5,
                            },
                            {},
                            undefined,
                          ]
                        }
                      >
                        <RNSVGSvgView
                          align="xMidYMid"
                          bbHeight="24"
                          bbWidth="24"
                          fill="none"
                          focusable={false}
                          height={24}
                          meetOrSlice={0}
                          minX={0}
                          minY={0}
                          style={
                            [
                              {
                                "backgroundColor": "transparent",
                                "borderWidth": 0,
                              },
                              {
                                "flex": 0,
                                "height": 24,
                                "width": 24,
                              },
                            ]
                          }
                          vbHeight={24}
                          vbWidth={24}
                          width={24}
                          xml="<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
    <g clip-path="url(#clip0_827_317)">
    <path d="M12 2C6.48 2 2 6.48 2 12C2 17.52 6.48 22 12 22C17.52 22 22 17.52 22 12C22 6.48 17.52 2 12 2ZM9.29 16.29L5.7 12.7C5.31 12.31 5.31 11.68 5.7 11.29C6.09 10.9 6.72 10.9 7.11 11.29L10 14.17L16.88 7.29C17.27 6.9 17.9 6.9 18.29 7.29C18.68 7.68 18.68 8.31 18.29 8.7L10.7 16.29C10.32 16.68 9.68 16.68 9.29 16.29Z" fill="#1FCC44"/>
    </g>
    <defs>
    <clipPath id="clip0_827_317">
    <rect width="24" height="24" fill="white"/>
    </clipPath>
    </defs>
    </svg>                
        "
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <RNSVGGroup
                            fill={null}
                            propList={
                              [
                                "fill",
                              ]
                            }
                          >
                            <RNSVGGroup
                              clipPath="clip0_827_317"
                              fill={
                                {
                                  "payload": 4278190080,
                                  "type": 0,
                                }
                              }
                            >
                              <RNSVGPath
                                d="M12 2C6.48 2 2 6.48 2 12C2 17.52 6.48 22 12 22C17.52 22 22 17.52 22 12C22 6.48 17.52 2 12 2ZM9.29 16.29L5.7 12.7C5.31 12.31 5.31 11.68 5.7 11.29C6.09 10.9 6.72 10.9 7.11 11.29L10 14.17L16.88 7.29C17.27 6.9 17.9 6.9 18.29 7.29C18.68 7.68 18.68 8.31 18.29 8.7L10.7 16.29C10.32 16.68 9.68 16.68 9.29 16.29Z"
                                fill={
                                  {
                                    "payload": 4280273988,
                                    "type": 0,
                                  }
                                }
                                propList={
                                  [
                                    "fill",
                                  ]
                                }
                              />
                            </RNSVGGroup>
                            <RNSVGDefs>
                              <RNSVGClipPath
                                fill={
                                  {
                                    "payload": 4278190080,
                                    "type": 0,
                                  }
                                }
                                name="clip0_827_317"
                              >
                                <RNSVGRect
                                  fill={
                                    {
                                      "payload": 4294967295,
                                      "type": 0,
                                    }
                                  }
                                  height="24"
                                  propList={
                                    [
                                      "fill",
                                    ]
                                  }
                                  width="24"
                                  x="0"
                                  y="0"
                                />
                              </RNSVGClipPath>
                            </RNSVGDefs>
                          </RNSVGGroup>
                        </RNSVGSvgView>
                      </View>
                    </View>
                    <Text
                      numberOfLines={0}
                      style={
                        {
                          "borderBottomWidth": 0,
                          "borderColor": undefined,
                          "color": "#4d5963",
                          "fontFamily": "Kanit-Regular",
                          "fontSize": 18,
                          "lineHeight": undefined,
                          "textAlign": undefined,
                        }
                      }
                    >
                      alma@mail.com
                    </Text>
                  </View>
                  <View
                    marginB-8={true}
                    style={
                      [
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        {},
                        {
                          "marginBottom": 8,
                        },
                        {},
                        undefined,
                      ]
                    }
                  >
                    <View
                      row={true}
                      style={
                        [
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          {},
                          {},
                          {
                            "flexDirection": "row",
                          },
                          {
                            "alignItems": "center",
                          },
                        ]
                      }
                    >
                      <Text
                        numberOfLines={0}
                        style={
                          {
                            "borderBottomWidth": 0,
                            "borderColor": undefined,
                            "color": "#216dab",
                            "fontFamily": "Kanit-Bold",
                            "fontSize": 22,
                            "lineHeight": undefined,
                            "textAlign": undefined,
                          }
                        }
                      >
                        phoneNumber
                      </Text>
                      <View
                        marginL-5={true}
                        style={
                          [
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            {},
                            {
                              "marginLeft": 5,
                            },
                            {},
                            undefined,
                          ]
                        }
                      >
                        <RNSVGSvgView
                          align="xMidYMid"
                          bbHeight="24"
                          bbWidth="24"
                          fill="none"
                          focusable={false}
                          height={24}
                          meetOrSlice={0}
                          minX={0}
                          minY={0}
                          style={
                            [
                              {
                                "backgroundColor": "transparent",
                                "borderWidth": 0,
                              },
                              {
                                "flex": 0,
                                "height": 24,
                                "width": 24,
                              },
                            ]
                          }
                          vbHeight={24}
                          vbWidth={24}
                          width={24}
                          xml="<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
    <g clip-path="url(#clip0_827_317)">
    <path d="M12 2C6.48 2 2 6.48 2 12C2 17.52 6.48 22 12 22C17.52 22 22 17.52 22 12C22 6.48 17.52 2 12 2ZM9.29 16.29L5.7 12.7C5.31 12.31 5.31 11.68 5.7 11.29C6.09 10.9 6.72 10.9 7.11 11.29L10 14.17L16.88 7.29C17.27 6.9 17.9 6.9 18.29 7.29C18.68 7.68 18.68 8.31 18.29 8.7L10.7 16.29C10.32 16.68 9.68 16.68 9.29 16.29Z" fill="#1FCC44"/>
    </g>
    <defs>
    <clipPath id="clip0_827_317">
    <rect width="24" height="24" fill="white"/>
    </clipPath>
    </defs>
    </svg>                
        "
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <RNSVGGroup
                            fill={null}
                            propList={
                              [
                                "fill",
                              ]
                            }
                          >
                            <RNSVGGroup
                              clipPath="clip0_827_317"
                              fill={
                                {
                                  "payload": 4278190080,
                                  "type": 0,
                                }
                              }
                            >
                              <RNSVGPath
                                d="M12 2C6.48 2 2 6.48 2 12C2 17.52 6.48 22 12 22C17.52 22 22 17.52 22 12C22 6.48 17.52 2 12 2ZM9.29 16.29L5.7 12.7C5.31 12.31 5.31 11.68 5.7 11.29C6.09 10.9 6.72 10.9 7.11 11.29L10 14.17L16.88 7.29C17.27 6.9 17.9 6.9 18.29 7.29C18.68 7.68 18.68 8.31 18.29 8.7L10.7 16.29C10.32 16.68 9.68 16.68 9.29 16.29Z"
                                fill={
                                  {
                                    "payload": 4280273988,
                                    "type": 0,
                                  }
                                }
                                propList={
                                  [
                                    "fill",
                                  ]
                                }
                              />
                            </RNSVGGroup>
                            <RNSVGDefs>
                              <RNSVGClipPath
                                fill={
                                  {
                                    "payload": 4278190080,
                                    "type": 0,
                                  }
                                }
                                name="clip0_827_317"
                              >
                                <RNSVGRect
                                  fill={
                                    {
                                      "payload": 4294967295,
                                      "type": 0,
                                    }
                                  }
                                  height="24"
                                  propList={
                                    [
                                      "fill",
                                    ]
                                  }
                                  width="24"
                                  x="0"
                                  y="0"
                                />
                              </RNSVGClipPath>
                            </RNSVGDefs>
                          </RNSVGGroup>
                        </RNSVGSvgView>
                      </View>
                    </View>
                    <Text
                      numberOfLines={0}
                      style={
                        {
                          "borderBottomWidth": 0,
                          "borderColor": undefined,
                          "color": "#4d5963",
                          "fontFamily": "Kanit-Regular",
                          "fontSize": 18,
                          "lineHeight": undefined,
                          "textAlign": undefined,
                        }
                      }
                    >
                      0812345674
                    </Text>
                  </View>
                  <View
                    marginB-8={true}
                    style={
                      [
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        {},
                        {
                          "marginBottom": 8,
                        },
                        {},
                        undefined,
                      ]
                    }
                  >
                    <Text
                      numberOfLines={0}
                      style={
                        {
                          "borderBottomWidth": 0,
                          "borderColor": undefined,
                          "color": "#216dab",
                          "fontFamily": "Kanit-Bold",
                          "fontSize": 22,
                          "lineHeight": undefined,
                          "textAlign": undefined,
                        }
                      }
                    >
                      companyName
                    </Text>
                    <Text
                      numberOfLines={0}
                      style={
                        {
                          "borderBottomWidth": 0,
                          "borderColor": undefined,
                          "color": "#4d5963",
                          "fontFamily": "Kanit-Regular",
                          "fontSize": 18,
                          "lineHeight": undefined,
                          "textAlign": undefined,
                        }
                      }
                    >
                      alma company
                    </Text>
                  </View>
                  <View
                    marginB-8={true}
                    style={
                      [
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        {},
                        {
                          "marginBottom": 8,
                        },
                        {},
                        undefined,
                      ]
                    }
                  >
                    <Text
                      numberOfLines={0}
                      style={
                        {
                          "borderBottomWidth": 0,
                          "borderColor": undefined,
                          "color": "#216dab",
                          "fontFamily": "Kanit-Bold",
                          "fontSize": 22,
                          "lineHeight": undefined,
                          "textAlign": undefined,
                        }
                      }
                    >
                      companyAddress
                    </Text>
                    <Text
                      numberOfLines={0}
                      style={
                        {
                          "borderBottomWidth": 0,
                          "borderColor": undefined,
                          "color": "#4d5963",
                          "fontFamily": "Kanit-Regular",
                          "fontSize": 18,
                          "lineHeight": undefined,
                          "textAlign": undefined,
                        }
                      }
                    >
                      pahlawan street
                    </Text>
                  </View>
                  <View
                    flexG-1={true}
                    marginB-30={true}
                    style={
                      [
                        undefined,
                        undefined,
                        {
                          "flexGrow": 1,
                        },
                        undefined,
                        {},
                        {
                          "marginBottom": 30,
                        },
                        {},
                        {
                          "justifyContent": "flex-end",
                        },
                      ]
                    }
                  >
                    <View
                      accessibilityState={
                        {
                          "busy": undefined,
                          "checked": undefined,
                          "disabled": undefined,
                          "expanded": undefined,
                          "selected": undefined,
                        }
                      }
                      accessibilityValue={
                        {
                          "max": undefined,
                          "min": undefined,
                          "now": undefined,
                          "text": undefined,
                        }
                      }
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        {
                          "alignItems": "center",
                          "backgroundColor": "#216dab",
                          "borderRadius": 6,
                          "elevation": 2,
                          "flexDirection": "row",
                          "gap": 0,
                          "justifyContent": "center",
                          "paddingHorizontal": 10,
                          "paddingVertical": 6,
                        }
                      }
                      testID="activate-button"
                    >
                      <Text
                        numberOfLines={0}
                        style={
                          {
                            "borderBottomWidth": 0,
                            "borderColor": undefined,
                            "color": "#f0f6fb",
                            "fontFamily": "Kanit-Medium",
                            "fontSize": 18,
                            "lineHeight": undefined,
                            "textAlign": undefined,
                          }
                        }
                      >
                        inactiveUser
                      </Text>
                      <View
                        style={
                          [
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            {
                              "position": "absolute",
                              "right": 20,
                            },
                          ]
                        }
                      />
                    </View>
                  </View>
                </View>
                <Modal
                  animationType="slide"
                  hardwareAccelerated={false}
                  onRequestClose={[Function]}
                  transparent={true}
                  visible={false}
                />
              </View>
            </RCTScrollView>
          </View>
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="UserDetail"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
